platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java

```diff
@@ -171,4 +186,36 @@ public static String formatException(final Throwable e) {
             throw new PlatformConstructionException("A problem occurred while creating the SocketFactory", e);
         }
     }
+
+    /**
+     * identifies the client on the other end of the socket using their signing certificate.
+     *
+     * @param certs        the list of all certificates in the public keystore
+     * @param peerInfoList List of known peers
+     * @return info of the identified peer
+     */
+    public static @Nullable PeerInfo identifyTlsPeer(
+            final @NonNull Certificate[] certs, final @NonNull List<PeerInfo> peerInfoList) {
+        PeerInfo matchingPair = null;
+        final X509Certificate agreementCert = (X509Certificate) certs[0];
+        final Set<PeerInfo> peers = peerInfoList.stream()
+                .filter(peerInfo -> ((X509Certificate) peerInfo.signingCertificate())
+                        .getSubjectX500Principal()
+                        .equals(agreementCert.getIssuerX500Principal()))
+                .collect(Collectors.toSet());
+        final Optional<PeerInfo> peer = peers.stream().findFirst();
+        if (peer.isEmpty()) {
+            rateLimitedLogger.warn(
+                    SOCKET_EXCEPTIONS.getMarker(),
+                    "Unable to find a matching peer with the presented certificate {}.",
+                    agreementCert);
+        } else {
+            rateLimitedLogger.info(
+                    SOCKET_EXCEPTIONS.getMarker(),
+                    "Found {} matching peers for the presented certificate",
+                    peers.size());
+            matchingPair = peer.get();
```

Comment:
this logic seems wrong. if we find a peer, it will log with a `SOCKET_EXCEPTIONS` marker. we should only log if there is more than one match. also, if this happens, we should use error level and `EXCEPTION` marker

Reply:
Looks like the logic is correct as per your comment, but the choice of logging isn't. Following up to clarify the markers.

Reply:
Just trying to follow the conversation.  There should be 3 rate limited logs here?  (1) error if there are no matches, (2) error if there are too many matches, (3) Info for successful single match?

Reply:
I don't know if my next comment is relevant here, but when using rate limited loggers, it is per message instance that needs to be rate limited independently of rate limiting other messages.     Suppose there are 3 log messages here for the different possible code-paths, and all 3 use the same rate limited logger.  Then the logging of a failure code-path will block the reporting of a successful code path until the time limit expires.  Vice versa, a report of success will block reporting of failures until the time limit expires.   All three code paths may need to have their own rate limited logger since we want to know the 1st instance of a message and only block out duplication of messages that have already been reported

Reply:
> Just trying to follow the conversation. There should be 3 rate limited logs here? (1) error if there are no matches, (2) error if there are too many matches, (3) Info for successful single match?

There should be two. Error if there are no matches, info if there are multiple matches.

Your second point is super helpful. I've just modified to add an additional rate limiter so one doesn't block the other. Unfortunately, I don't have a mechanism for testing/observing this behavior locally.